examples/xlio_ultra_api_ping_pong.c

```diff
@@ -972,9 +832,451 @@ static void process_received_message(xlio_socket_t sock, const char *msg, size_t
+    if (!g_state.pd) {
+        g_state.pd = g_state.api->xlio_socket_get_pd(sock);
+        if (!g_state.pd) {
+            fprintf(stderr,
+                    "Cannot register TX buffers: failed to get protection domain from socket\n");
+            return -1;
+        }
+    }
+
+    /* Register the entire memory block */
+    g_state.mr_buf = ibv_reg_mr(g_state.pd, g_state.tx_buffers, total_size, IBV_ACCESS_LOCAL_WRITE);
+    if (!g_state.mr_buf) {
+        fprintf(stderr, "Failed to register TX buffer memory\n");
+        return -1;
+    }
+
+    printf("TX buffer memory registered\n");
+
+    return 0;
+}
+
+/*
+ * Get TX buffer from pool
+ * =======================
+ * Returns buffer pointer on success, NULL if pool is empty.
+ */
+static char *get_tx_buffer(void)
+{
+    char *buffer_ptr;
+
+    /* Check if pool has available buffers */
+    if (g_state.pool_top < 0) {
+        return NULL;
+    }
+
+    /* Pop buffer pointer from stack */
+    buffer_ptr = g_state.pool_stack[g_state.pool_top--];
+
+    return buffer_ptr;
+}
+
+/*
+ * Return TX buffer to pool
+ * ========================
+ */
+static void return_tx_buffer(char *buffer_ptr)
+{
+    size_t offset;
+    size_t total_size = TX_BUFFER_COUNT * TX_BUFFER_SIZE;
+
+    /* Validate that buffer pointer belongs to our memory block */
+    if (buffer_ptr < g_state.tx_buffers || buffer_ptr >= g_state.tx_buffers + total_size) {
+        fprintf(stderr, "Invalid buffer pointer %p (not in memory block range %p-%p)\n",
+                (void *)buffer_ptr, (void *)g_state.tx_buffers,
+                (void *)(g_state.tx_buffers + total_size));
+        return;
+    }
+
+    /* Validate that buffer pointer is properly aligned to buffer boundaries */
+    offset = buffer_ptr - g_state.tx_buffers;
+    if (offset % TX_BUFFER_SIZE != 0) {
+        fprintf(stderr, "Invalid buffer pointer %p (not aligned to buffer boundary)\n",
+                (void *)buffer_ptr);
+        return;
+    }
+
+    /* Check for stack overflow */
+    if (g_state.pool_top >= TX_BUFFER_COUNT - 1) {
+        fprintf(stderr, "TX buffer pool stack overflow (double-free of buffer %p?)\n",
+                (void *)buffer_ptr);
+        return;
+    }
+
+    /* Push buffer pointer back onto stack */
+    g_state.pool_stack[++g_state.pool_top] = buffer_ptr;
+}
+
+/*
+ * Cleanup TX buffer pool
+ * ======================
+ * Deregisters memory and frees all allocated resources.
+ */
+static void cleanup_tx_buffer_pool(void)
+{
+    if (g_state.mr_buf) {
+        ibv_dereg_mr(g_state.mr_buf);
+        g_state.mr_buf = NULL;
+    }
+
+    if (g_state.tx_buffers) {
+        free(g_state.tx_buffers);
+        g_state.tx_buffers = NULL;
+    }
+
+    if (g_state.pool_stack) {
+        free(g_state.pool_stack);
+        g_state.pool_stack = NULL;
+    }
+
+    g_state.pool_top = -1;
+}
+
+/*
+ * =================
+ * Utility Functions
+ * =================
+ */
+
+/*
+ * Return maximum of two size_t values
```

Comment:
I read online and seen why it's needed, please enhacne this to:
`/*
 * Return maximum of two size_t values
 * Standard C has no max() for size_t, so we provide our own.
 */`

Reply:
done